### ISSUE TITLE
FreeBSD warning in stubs.cpp

### ISSUE THREAD

**Issue by chief1983** [scp-fs2open/fs2open.github.com#1138]
```
[ 90%] Building CXX object code/CMakeFiles/code.dir/windows_stub/stubs.cpp.o
/usr/home/chief1983/fs2_open_git/code/windows_stub/stubs.cpp:80:20: warning: comparison of integers of different signs: 'int' and 'unsigned long' [-Wsign-compare]
        for (int i = 1; i < numstrings; i++)
                        ~ ^ ~~~~~~~~~~
1 warning generated.
```

Not sure how it got past other compilers, unless they don't end up with 'unsigned long' from the auto type numstrings is defined with.

**Comment by asarium** [comment]
On Linux the signature is `int backtrace (void **, int)` so there is no signed/unsigned conflict on the CI systems. This could be fixed in a cross-platform way by always using `size_t` but that may cause issues on Linux where the backtrace functions use `int` instead of `size_t`.

**Comment by chief1983** [comment]
Is Mac behaving more like Linux than FreeBSD in this case?  Because since it is using clang and is a BSD derivative I would have expected it to be more like FreeBSD.  But I got no such warning on my latest Mac builds.

**Comment by asarium** [comment]
According to the [Apple Documentation](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man3/backtrace.3.html) it uses the `int` version of the functions.

**Comment by Goober5000** [comment]
A little farther down in the same file, we have this:
```
	for (auto i = 0; i < numstrings; ++i) {
```

Can the offending line be fixed by using `auto`?

As an aside, I don't know if this is a CMake thing, but `stubs.cpp` is not part of the Visual Studio project, even though `config.h` in the same folder is.

**Comment by asarium** [comment]
`auto` wouldn't work since `i` would still be an `int` in that case. However, something like `decltype(numstrings)` should work since then it would automatically use the right type.

**Comment by Goober5000** [comment]
Sounds good.  I opened a new PR.

And, incidentally, I figured out that `stubs.cpp` is not included because it's not a Windows file.

**Comment by Goober5000** [comment]
Fixed in #2107
